zebra-network/src/peer/handshake.rs

```diff
@@ -277,6 +282,18 @@ where
                         msg
                     }
                 })
+                .then(move |msg| {
+                    let inv_collector = inv_collector.clone();
+                    async move {
+                        if let Ok(Message::Inv(hashes)) = &msg {
+                            if hashes.len() == 1 {
```

Comment:
Let's put a brief comment (possibly referencing the RFC) explaining why we check len == 1 here, since without that context it's a little unintuitive.